### DISCUSSION TITLE
Export paths of save files on the system and of those not backed and disabled

### DISCUSSION THREAD

**Discussion by tekert** [mtkennerly/ludusavi#238] [Ideas]
I use a backup program that uses .gitignore like files for file exclusion.
The problem is that my save "repo" across all location is of several gigabyes,
I want to exclude those paths from the system data backup to exclude duplicated backups since ludusavi backups are grouped with the system-data backups when uploading to the cloud.

My idea is to create a command to dumps all file paths, one path per line, (.gitignore compatible) of all the backedup saves on the system.
I would be cool to also list paths of games with disabled backups by setting up flags.

edit:
I'm thinking to also filter the list of paths returned by base path, for example, to give me all paths that belong to a parent path, it will make it easier to setup custom scripts since rust may parse these paths faster.

**Comment by mtkennerly** [comment]
Hi! You could write a script to do this by parsing the output from `ludusavi restore --preview --api`. That will give you a list of all saves from your latest backup for each game. If you have redirects configured, you'd just need to check the `originalPath` field for the location on disk.

If you also want to find saves that haven't been backed up yet, you can use `ludusavi backup --preview --api` and filter on `"change": "New"`. Disabled saves will be marked with `"ignored": true`.

**Reply by tekert** [reply]
Oh, Thanks, exelent, totally missed the `--api` returning a json
`ludusavi restore --preview --api` works perfect, it's fast and returns all backedup paths.

The problem i'm having is getting the ignored games from ludusavi in a faster way because `ludusavi backup --preview --api` takes 4min, and sometimes when a ssd is scanned goes 100% cpu use, and saturates the 6tb hdd drive for 2min. Since this would be automated at regultar intervals (once the OS cached these files, the same command goes down to 50sec) but i would like to damper the initial query.
It would be cool to get the list of ignored games like when you open ludusavi.exe GUI it already has a list of ignored games without scanning anything, it is possible to get the paths for those? like scanning only the ignored games for paths.

**Reply by mtkennerly** [reply]
That makes sense \:) I've just added some more options to the `find` command. In the next release, you'll be able to do this:

```console
# Games that are disabled
$ ludusavi find --disabled
Pathologic
Valheim

# Games that are disabled for backup specifically
$ ludusavi find --disabled --backup
Pathologic

# Games that are enabled, but have some disabled saves
$ ludusavi find --partial
A Custom Steam Shortcut
The Legend of Heroes: Trails in the Sky

# Combine and pipe it (example with Bash)
$ { ludusavi find --disabled && ludusavi find --partial; } | uniq | ludusavi backup --preview --api
{
  "overall": { ... },
  "games": {
    "A Custom Steam Shortcut": { ... },
    "Pathologic": { ... }
    "The Legend of Heroes: Trails in the Sky": { ... },
    "Valheim": { ... }
  }
}
```

**Reply by tekert** [reply]
Oh ;), exelent, I was already trying to parse config.yaml to get the list of ignored and partiral games in parts and then parse json , but parsing yaml on powershell is a bit verbose for my taste, that command would be much better, Thanks!
